docs/DesignDocs/Enabledebugmalloc.md

```diff
@@ -0,0 +1,154 @@
+Proposal to enable debug malloc for end users
+===
+
+# Motivation
```

Comment:
Per the SIG arch meeting, I think it would be useful to discuss the use cases that we expect a developer to use this feature for, and to move the description of the intended logic into the Design section below. 

It sounds like the feature is designed to address at least two major use cases:

- A developer wants to track all allocations from the start of the enclave until it is terminated and determine if any allocations were leaked.
- A developer wants to track allocations in a specific region of the enclave app code that they can specify at dev time.

Reply:
Yes exactly as you described. Thanks!